geode-core/src/distributedTest/java/org/apache/geode/ClusterSSLDUnitTest.java

```diff
@@ -0,0 +1,232 @@
+      verifyCreatedEntry(VM.getVM(i));
+    }
+    performUpdate(VM.getVM(1));
+    for (int i = 1; i <= NUM_SERVERS; i++) {
+      verifyUpdatedEntry(VM.getVM(i));
+    }
+  }
+
+  @Test
+  public void createEntryWithThreadOwnedSocketsAndBigMessage() throws Exception {
+    int locatorPort = createLocator(VM.getVM(0));
+    for (int i = 1; i <= NUM_SERVERS; i++) {
+      createSSLEnabledCacheAndRegion(VM.getVM(i), locatorPort, false);
+    }
+    performCreateWithLargeValue(VM.getVM(1));
+    for (int i = 1; i <= NUM_SERVERS; i++) {
+      verifyCreatedEntry(VM.getVM(i));
+    }
+  }
+
+  @Test
+  public void performARollingUpgrade() throws Exception {
+    List<String> testVersions = VersionManager.getInstance().getVersionsWithoutCurrent();
+    Collections.sort(testVersions);
+    String testVersion = testVersions.get(testVersions.size() - 1);
+
+    // create a cluster with the previous version of Geode
+    VM locatorVM = Host.getHost(0).getVM(testVersion, 0);
+    VM server1VM = Host.getHost(0).getVM(testVersion, 1);
+    int locatorPort = createLocator(locatorVM);
+    createSSLEnabledCacheAndRegion(server1VM, locatorPort, true);
+    performCreate(VM.getVM(1));
+
+    // roll the locator to the current version
+    locatorVM.invoke("stop locator", () -> Locator.getLocator().stop());
+    locatorVM = Host.getHost(0).getVM(VersionManager.CURRENT_VERSION, 0);
+    locatorVM.invoke("roll locator to current version", () -> {
+      // System.setProperty("javax.net.debug", "all");
+      Properties props = getDistributedSystemProperties();
+      // locator must restart with the same port so that it reconnects to the server
+      GeodeAwaitility.await().atMost(15, TimeUnit.SECONDS)
+          .until(() -> Locator.startLocatorAndDS(locatorPort, new File(""), props) != null);
+      assertThat(Locator.getLocator().getDistributedSystem().getAllOtherMembers().size())
+          .isGreaterThan(0);
+    });
+
+    // start server2 with current version
+    VM server2VM = Host.getHost(0).getVM(VersionManager.CURRENT_VERSION, 2);
+    createSSLEnabledCacheAndRegion(server2VM, locatorPort, true);
+
+    // roll server1 to the current version
+    server1VM.invoke("stop server1", () -> {
+      CacheFactory.getAnyInstance().getDistributedSystem().disconnect();
+    });
+    server1VM = Host.getHost(0).getVM(VersionManager.CURRENT_VERSION, 1);
+    createSSLEnabledCacheAndRegion(server1VM, locatorPort, true);
+
+
+    verifyCreatedEntry(server1VM);
+    verifyCreatedEntry(server2VM);
+  }
+
+  private void createSSLEnabledCacheAndRegion(VM memberVM, int locatorPort,
+      boolean conserveSockets) {
+    memberVM.invoke("start cache and create region", () -> {
+      Cache cache = createCache(locatorPort, conserveSockets);
+      cache.createRegionFactory(RegionShortcut.REPLICATE).create(regionName);
+    });
+  }
+
+
+  private void performCreate(VM memberVM) {
+    memberVM.invoke("perform create", () -> CacheFactory.getAnyInstance()
```

Comment:
I'm glad you mentioned this because I was used to the JUnit4CacheTestCase behavior of disabling shutdown of the DistributedSystem when the cache is closed.  The new DistributedRule doesn't appear to do this.

Reply:
I removed use of CacheFactory.getAnyInstance().